src/server/utils/settings-manager/settings-manager.ts

```diff
@@ -126,23 +126,27 @@ export class SettingsManager {
     return this.timeMonitor.timekeeper;
   }
 
-  getSettings(opts: GetSettingsOptions = {}): Promise<AppSettings> {
-    let currentWork = this.currentWork;
+  handleSettingsTask(task: Promise<void>, opts: GetSettingsOptions = {}): Promise<AppSettings> {
+    const timeoutMs = opts.timeout || this.initialLoadTimeout;
+    if (timeoutMs === 0) {
+      return task.then(() => this.appSettings);
+    }
+    return Promise.race([task, timeout(timeoutMs)])
+      .catch(() => {
+        this.logger.error("Settings load timeout hit, continuing");
```

Comment:
error or warn? 
put timeout value in the message, will be more descriptive

Reply:
Makes sense!